src/test/java/uk/gov/hmcts/ccd/domain/types/DocumentValidatorTest.java

```diff
@@ -53,10 +55,41 @@ public static void setUpClass() {
         BaseType.setCaseDefinitionRepository(definitionRepository);
     }
 
+    @Test
+    public void shouldValidateIfPortsAreSpecifiedAndMatch() {
+        final DocumentValidator validatorWithPort = buildDocumentValidator("https://ng.reform.hmcts.net:6789");
+
+        final ObjectNode data = createDoc(DOCUMENT_URL_WITH_PORT);
+        final List<ValidationResult> validDocumentUrlResult = validatorWithPort.validate(DOCUMENT_FIELD_ID,
+            data, caseField);
+        assertThat(validDocumentUrlResult, empty());
+    }
+
+    @Test
+    public void shouldNotValidateIfPortsAreSpecifiedAndNotMatch() {
+        final DocumentValidator validatorWithPort = buildDocumentValidator("https://ng.reform.hmcts.net:7789");
+
+        final ObjectNode data = createDoc(DOCUMENT_URL_WITH_PORT);
+        final List<ValidationResult> validDocumentUrlResult = validatorWithPort.validate(DOCUMENT_FIELD_ID,
+            data,
+            caseField);
+        assertThat(validDocumentUrlResult, hasSize(1));
+        assertThat(validDocumentUrlResult.get(0).getErrorMessage(),
+            is(DOCUMENT_URL_WITH_PORT + " does not match Document Management domain or expected URL path"));
+    }
+
+    private DocumentValidator buildDocumentValidator(final String url) {
+        final ApplicationParams ap = mock(ApplicationParams.class);
+        when(ap.getValidDMDomain()).thenReturn(url);
+        return new DocumentValidator(ap);
+    }
+
     @Before
     public void setUp() throws Exception {
-        validator = new DocumentValidator();
-
+        final ApplicationParams applicationParams = mock(ApplicationParams.class);
+//        when(applicationParams.getValidDMDomain()).thenReturn("https://dm.reform.hmcts.net");
```

Comment:
Why is this commented out?

Reply:
Missed it, was testing out something locally 👍

Reply:
@mrganeshraja are you uncommenting this out?